### ISSUE TITLE
[Bug]: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'method')

### ISSUE THREAD

**Issue by PeterYinusa** [MetaMask/metamask-extension#15526]
### Describe the bug

Sometimes when switching networks the following error is thrown, the data object in the error sometimes does not have a request property.

{"code":-32603,"data":{"code":-32602,"message":"Unknown block number"}}

https://github.com/MetaMask/metamask-extension/blob/769b16eaa9189f58761a044e8039e9403884cb97/app/scripts/lib/account-tracker.js#L252

### Steps to reproduce

1. Switch networks back and forth from Ethereum Mainnet to Optimism
2. Check the background console

### Error messages or log output

```shell
background-2.js:15 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'method')
  at r.default._updateAccount (background-2.js:15:6755)
  at async Promise.all (index 0)
  at async r.default._updateAccounts (background-2.js:15:6531)
```

### Version

10.19.0

### Browser

Chrome

### Operating system

MacOS

**Comment by PeterYinusa** [comment]
Similar issue below:

{"code":-32000,"message":"header not found"}

```
background-2.js:15 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'request')
  at r.default._updateAccount (background-2.js:15:6747)
  at async Promise.all (index 4)
```

**Comment by mabioIa** [comment]
During testing, it resolves pretty nicely when optionally chaining the `request` property, kindly [confirm](https://github.com/mabioIa/metamask-extension). Created a [pull request](https://github.com/MetaMask/metamask-extension/pull/15563) if satisfied with the solution.

**Comment by danjm** [comment]
Closed by #15889
